components/src/Tooltip/Tooltip.js

```diff
@@ -0,0 +1,282 @@
+        },
+        "&:after": {
+          borderColor: `${tooltipStyles.backgroundColor} transparent transparent transparent`,
+          borderWidth: `${theme.space.x1} ${theme.space.x1} 0 ${theme.space.x1}`,
+          left: `-${theme.space.half}`,
+        },
+      });
+    case "right":
+      return ({
+        height: theme.space.x1,
+        left: 0,
+        marginLeft: `-${theme.space.x1}`,
+        width: theme.space.x1,
+        "&:before": {
+          borderColor: `transparent ${tooltipStyles.borderColor} transparent transparent`,
+          borderWidth: `${theme.space.x1} ${theme.space.x1} ${theme.space.x1} 0`,
+          top: `-${theme.space.half}`,
+        },
+        "&:after": {
+          borderColor: `transparent ${tooltipStyles.backgroundColor} transparent transparent`,
+          borderWidth: `${theme.space.x1} ${theme.space.x1} ${theme.space.x1} 0`,
+          left: "2px",
+          top: `-${theme.space.half}`,
+        },
+      });
+    case "left":
+      return ({
+        height: theme.space.x1,
+        marginRight: `-${theme.space.x1}`,
+        right: 0,
+        width: theme.space.x1,
+        "&:before": {
+          borderColor: `transparent transparent transparent ${tooltipStyles.borderColor}`,
+          borderWidth: `${theme.space.x1} 0 ${theme.space.x1} ${theme.space.x1}`,
+          top: `-${theme.space.half}`,
+        },
+        "&:after": {
+          borderColor: `transparent transparent transparent ${tooltipStyles.backgroundColor}`,
+          borderWidth: `${theme.space.x1} 0 ${theme.space.x1} ${theme.space.x1}`,
+          left: "-2px",
+          top: `-${theme.space.half}`,
+        },
+      });
+    default:
+      return ({});
+  }
+};
+
+const Arrow = styled.div(
+  {
+    height: theme.space.x1,
+    position: "absolute",
+    width: theme.space.x1,
+    margin: "12px",
+    "&:before": {
+      borderStyle: "solid",
+      content: "''",
+      display: "block",
+      height: 0,
+      margin: "auto",
+      position: "absolute",
+      width: 0,
+    },
+    "&:after": {
+      borderStyle: "solid",
+      content: "''",
+      display: "block",
+      height: 0,
+      margin: "auto",
+      position: "absolute",
+      width: 0,
+    },
+  },
+  ({ dataPlacement }) => ({
+    ...getArrowPosition(dataPlacement),
+  })
+);
+
+class Tooltip extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      open: false,
+    };
+    this.escFunction = this.escFunction.bind(this);
+    this.hideTooltip = this.hideTooltip.bind(this);
+    this.showTooltip = this.showTooltip.bind(this);
+  }
+
+  componentWillMount() {
+    document.addEventListener("keydown", this.escFunction, false);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener("keydown", this.escFunction, false);
+  }
+
+  clearScheduled = () => {
+    clearTimeout(this.hideTimeout);
+    clearTimeout(this.showTimeout);
+  };
+
+  getTooltipProps = () => ({
```

Comment:
Would it make sense to make the name more specific?
e.g. `tooltipEventHandlers`